text/0015-gas-payments.md

```diff
@@ -0,0 +1,90 @@
+# Casper Payments: Gas Payment Concepts
+
+## High level view
+This is the first of a few CEPs on Casper Payment model. 
+The motivation for this work is to make Casper much more user and dev 
+friendly than it is now. There are a few topics here, that mix. The change 
+to one of them might cause the domino effect on others, but we'll 
+try to chunk them into logical units.
+
+#### Token Standard 
+Most of the systems built on top of smart contracting platforms use tokens 
+and it is reasonable to put a lot of effort in designing a good token standard 
+or discover gaps in core functionalities, that should be fixed before 
+the Mainnet. The most popular token standard right now is the ERC20. We have 
+implemented its Casper version here https://github.com/casperlabs/erc20.
+We think we can do better and offer our future community a token standard 
+that avoids the "approve and call" pitfall. 
+This work will be the subject of future CEP.
+
+#### CSPR Interface
```

Comment:
Can you elaborate on what exactly is problematic about ETH not implementing a good token standard? What constitutes a good token standard? Is there an example among existing ERC standards that you would consider good?

Reply:
> Can you elaborate on what exactly is problematic about ETH not implementing a good token standard? 

Uniswap v1 is a good example. Because the Ether doesn't implement ERC20 interface they had to implement more exchange methods:
* Ether to ERC20
* ERC20 to Ether
* ERC20 to ERC20

If Ether would implement ERC20, then the last one would be enough. The solution is to put Ether inside the ERC20 - https://weth.io/ But that creates the situation, that some platforms expects WETH, because that was easier for them to write, and some expects plain Ether, because they optimize for gas cost (plain Ether transferts are cheaper).

> What constitutes a good token standard?

I think the adoption is the best indicator. That's why iterating on top of the ERC20 is a good idea.

>  Is there an example among existing ERC standards that you would consider good?

ERC20. I know, I'm boring :smile: